src/App.vue

```diff
@@ -1,51 +1,121 @@
 <template>
-  <div id="nav">
-    <router-link to="/">Home</router-link> |
-    <router-link v-if="!isLoggedIn" to="/login/">Login</router-link>
-    <a href="#" v-if="isLoggedIn" @click="logout"> Logout</a>
+  <div id="header">
+    <div id="nav">
+      <router-link to="/">{{ $t("nav.home") }}</router-link> |
+      <router-link v-if="!isLoggedIn" to="/login/">{{ $t("nav.login") }}</router-link>
+      <a href="#" v-if="isLoggedIn" @click="logout">{{ $t("nav.logout") }}</a>
+    </div>
+    <div class="overlay">
+      <LocaleSwitcher id="locale"></LocaleSwitcher>
+    </div>
   </div>
   <router-view />
 </template>
 
 <script>
+import LocaleSwitcher from "./components/LocaleSwitcher.vue";
+import axios from "axios";
+
 export default {
+  components: {
+    LocaleSwitcher,
+  },
+  created() {
+    if (this.isLoggedIn && !this.hasLocalUserConfig) {
+      // fetch user config for logged in users if not already present
+      this.saveLocalUserConfig();
+    }
+    // set locale based on their config
+    this.setLocale();
+  },
   methods: {
     logout() {
       this.$store.dispatch("logout").then(this.$router.push("/login/"));
     },
+    saveLocalUserConfig() {
+      axios
+        .get(
+          process.env.VUE_APP_BACKEND +
+            process.env.VUE_APP_BACKEND_USER_CONFIG +
+            "?user-id=" +
+            localStorage.phone
+        )
+        .then((response) => {
+          this.$store.dispatch("saveConfig", {
+            config: JSON.stringify(response.data), // save user config locally
+          });
+        });
+    },
+    setLocale() {
+      var redirectId = setInterval(() => {
+        // for some reason, the localStorage.config has the correct value
+        // but when fetching the same through Vuex, it gives null
+        // upon refreshing, the fetch from Vues starts working fine
```

Comment:
Because of this: https://stackoverflow.com/questions/55356977/vuex-state-not-updating-from-local-storage#:~:text=directly%20(for%20example%20clicking%20a,the%20key%20doesn't%20exist.